### ISSUE TITLE
Resolution error not displayed when previous verification errors occurred

### ISSUE THREAD

**Issue by MikaelMayer** [dafny-lang/dafny#1926]
Forget about the gutter on the left, just look at the error message:
![image](https://user-images.githubusercontent.com/3601079/159571856-e1674b63-905e-4f4c-825c-facc2c72b88c.png)

This is a verification error taken from a previous execution, which appears on hover on all the expression _except on the opening square bracket_, where it adds the resolution error afterwards:
![image](https://user-images.githubusercontent.com/3601079/159572176-95fea00e-c824-4af0-a09a-f18fe801e4a7.png)

**Comment by keyboardDrummer** [comment]
Could you add code to make it easier to reproduce? Add possibly add the LSP server response that's faulty.

**Comment by MikaelMayer** [comment]
Here is some minimal code that makes it easy to reproduce:

Initial code: Make sure you save it, or diagnostics are not relocated.
```dafny
function schedule(x: int): int
predicate P(x: int)

method Main() {
  var i := 0;
  ghost var x :| P(x);
}
```
Which correctly displays an error on `:|`
> cannot establish the existence of LHS values that satisfy the such-that predicate - Verifier[6, 15]

Now, place the caret between the `:` and the `|`, and insert the following text. Speed does not matter:

` schedule[i]`

Now, press the DEL key 6 times.
The final program is the following:

```dafny
function schedule(x: int): int
predicate P(x: int)

method Main() {
  var i := 0;
  ghost var x := schedule[i];
}
```
And you reproduce the same error as above, meaning a completely outdated verification error that spans `:= schedule[i]` and a tiny but important resolution error that spans only `[`

Note that there is a case even more critical which behaves badly:
```
predicate P(x: int)

method M1() {
  ghost var x :| P(x);
}
method M2() {
  ghost var x :| P(x);
}
method M3() {
  ghost var x :| P(x);
}
method M4() {
  ghost var x :| P(x);
}
method M5() {
  ghost var x :| P(x);
}
method M6() {
  ghost var x :| P(x);
}
method M7() {
  ghost var x :| P(x);
}
```
Here you get a lot of verification errors. Let's say you first want to fix `M7()` but insert a parse error after the final P(x).
You won't see it in the diagnostics window because it's below all the other 7 verifications errors. It happened to me last week and I had a crazy 60 verification errors while migrating some code, and without the branch `verification-diagnostics` I would have thought the server had crashed on my file.

In the branch `verification-diagnostics`, to mitigate that effect, I did some changes so that it would highlight the resolution error and not the verification error (which would only get the underline red squiggle)
![image](https://user-images.githubusercontent.com/3601079/159720885-30056405-a416-425d-9c0a-41f3d6c1170e.png)

I'll open a fix in a PR in the next minutes
-

**Comment by keyboardDrummer** [comment]
If you mark the old verification diagnostics as warnings, I suggest also prefixing them with "Stale: " to help the customer understand why they're only warnings

**Comment by keyboardDrummer** [comment]
> You won't see it in the diagnostics window because it's below all the other 7 verifications errors

Do you know what determines the order? Is that the order in the file?

**Comment by MikaelMayer** [comment]
After trying this for some time, I don't know what happens, but it keeps duplicating the warnings in some cases so I'll drop this "fix" in my PR and we need to think further.
```dafny
// The document loader failed to create a new symbol table. Since we'd still like to provide
        // features such as code completion and lookup, we re-locate the previously resolved symbols
        // according to the change.
        return WithRealtimeDiagnosticsPublished(newDocument with {
          SymbolTable = relocator.RelocateSymbols(oldDocument.SymbolTable, documentChange, CancellationToken.None),
          OldVerificationDiagnostics = migratedVerificationDiagnotics.Select(diagnostic => diagnostic with {
            Severity = DiagnosticSeverity.Warning,
            Message = diagnostic.Message.StartsWith("Stale: ") ? diagnostic.Message : "Stale: " + diagnostic.Message
          }).ToList(),
          VerificationNodeDiagnostic = migratedVerificationNodeDiagnostics
        });
```

**Comment by keyboardDrummer** [comment]
@MikaelMayer is this issue this ongoing?

**Comment by MikaelMayer** [comment]
Yes, every single day I use Dafny. I learned to do workarounds, like looking at the diagnostics windows.
But I would really like that warnings that overlap with errors would actually point to the error directly, because that's the most important element. Perhaps widen
